consensus/istanbul/qbft/core/request.go

```diff
@@ -42,11 +45,30 @@ func (c *core) handleRequest(request *Request) error {
 
 	c.current.pendingRequest = request
 	if c.state == StateAcceptRequest {
-		// Start ROUND-CHANGE timer
-		c.newRoundChangeTimer()
+		c.newRoundMutex.Lock()
+		defer c.newRoundMutex.Unlock()
+
+		if c.newRoundTimer != nil {
+			c.newRoundTimer.Stop()
+			c.newRoundTimer = nil
+		}
 
-		// Send PRE-PREPARE message to other validators
-		c.sendPreprepareMsg(request)
+		delay := time.Duration(0)
+
+		block, ok := request.Proposal.(*types.Block)
+		if ok && len(block.Transactions()) == 0 { // if empty block
+			delay = -time.Since(time.Unix(int64(block.Time()), 0)) + time.Duration(c.config.EmptyBlockPeriod-c.config.BlockPeriod)*time.Second
```

Comment:
I don't have the parent block information at this place, block time should have the parent block time + block period as value.
I am checking for the negative value.

Reply:
no problem for negative value or 0, it is executed immediately
https://go.dev/play/p/QB80T3p4KRa

Reply:
delta from block time and now should be 0, I will check

Reply:
```
quorum-examples-node1-1  | INFO [07-08|14:58:32.176] qbft block                               time=1657292312 since=176.173958ms
quorum-examples-node1-1  | INFO [07-08|14:58:44.035] qbft block                               time=1657292324 since=35.466547ms
quorum-examples-node1-1  | INFO [07-08|14:58:53.048] qbft block                               time=1657292333 since=48.515009ms
quorum-examples-node1-1  | INFO [07-08|14:59:02.044] qbft block                               time=1657292342 since=44.105264ms
quorum-examples-node1-1  | INFO [07-08|14:59:11.016] qbft block                               time=1657292351 since=16.593976ms
quorum-examples-node1-1  | INFO [07-08|14:59:20.078] qbft block                               time=1657292360 since=78.740772ms
quorum-examples-node1-1  | INFO [07-08|14:59:29.062] qbft block                               time=1657292369 since=62.659776ms
quorum-examples-node1-1  | INFO [07-08|14:59:38.058] qbft block                               time=1657292378 since=58.693822ms
quorum-examples-node1-1  | INFO [07-08|14:59:47.045] qbft block                               time=1657292387 since=45.062118ms
quorum-examples-node1-1  | INFO [07-08|14:59:56.047] qbft block                               time=1657292396 since=47.224956ms
quorum-examples-node1-1  | INFO [07-08|15:00:05.019] qbft block                               time=1657292405 since=19.969544ms
quorum-examples-node1-1  | INFO [07-08|15:00:14.035] qbft block                               time=1657292414 since=35.883881ms
quorum-examples-node1-1  | INFO [07-08|15:00:23.028] qbft block                               time=1657292423 since=28.577886ms
quorum-examples-node1-1  | INFO [07-08|15:00:32.038] qbft block                               time=1657292432 since=38.90014ms
```
so as it is in seconds, there is no need to add time.Since(time.Unix(int64(block.Time()), 0)), it is negligeable, and we are back to my original formula